File: src/main/java/net/lomeli/lomlib/client/addon/jei/anvil/FluidAnvilRecipeWrapper.java
package net.lomeli.lomlib.client.addon.jei.anvil;

import com.google.common.collect.ImmutableList;
import mezz.jei.api.recipe.IRecipeWrapper;
import mezz.jei.api.recipe.wrapper.ICraftingRecipeWrapper;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import net.minecraft.client.Minecraft;
import net.minecraft.item.ItemStack;

import net.minecraftforge.fluids.FluidStack;

import net.lomeli.lomlib.client.addon.jei.BlankRecipeWrapper;
import net.lomeli.lomlib.core.recipes.FluidAnvilRecipe;

public class FluidAnvilRecipeWrapper implements IRecipeWrapper {
    @Nonnull
    private final FluidAnvilRecipe recipe;

    public FluidAnvilRecipeWrapper(FluidAnvilRecipe recipe) {
        this.recipe = recipe;
    }

    @Override
    public List getInputs() {
        return Collections.singletonList(recipe.recipeInputs());
    }

    @Override
    public List<ItemStack> getOutputs() {
        return Collections.singletonList(recipe.recipeOutput());
    }

    @Override
    public void drawAnimations(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight) {
    }

    @Nullable
    @Override
    public List<String> getTooltipStrings(int mouseX, int mouseY) {
        return null;
    }

    @Override
    public List<FluidStack> getFluidInputs() {
        return ImmutableList.of();
    }

    @Override
    public List<FluidStack> getFluidOutputs() {
        return ImmutableList.of();
    }

    @Override
    public void drawInfo(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight) {

    }

    @Override
    public void drawInfo(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight, int mouseX, int mouseY) {

    }

    @Override
    public boolean handleClick(@Nonnull Minecraft minecraft, int mouseX, int mouseY, int mouseButton) {
        return false;
    }
}

File: src/main/kotlin/net/lomeli/lomlib/client/addon/jei/anvil/FluidAnvilRecipeWrapper.kt
package net.lomeli.lomlib.client.addon.jei.anvil

import com.google.common.collect.ImmutableList
import mezz.jei.api.ingredients.IIngredients
import mezz.jei.api.recipe.IRecipeWrapper
import java.util.Collections

import net.minecraft.client.Minecraft
import net.minecraft.item.ItemStack

import net.minecraftforge.fluids.FluidStack

import net.lomeli.lomlib.core.recipes.FluidAnvilRecipe

class FluidAnvilRecipeWrapper(private val recipe: FluidAnvilRecipe) : IRecipeWrapper {

    fun getInputs(): List<*> {
        return listOf(recipe.recipeInputs())
    }

    fun getOutputs(): List<ItemStack> {
        return listOf(recipe.recipeOutput())
    }

    fun drawAnimations(minecraft: Minecraft, recipeWidth: Int, recipeHeight: Int) {
    }

    override fun getTooltipStrings(mouseX: Int, mouseY: Int): List<String>? {
        return null
    }

    fun getFluidInputs(): List<FluidStack> {
        return ImmutableList.of<FluidStack>()
    }

    fun getFluidOutputs(): List<FluidStack> {
        return ImmutableList.of<FluidStack>()
    }

    override fun drawInfo(minecraft: Minecraft, recipeWidth: Int, recipeHeight: Int, mouseX: Int, mouseY: Int) {
    }

    override fun handleClick(minecraft: Minecraft, mouseX: Int, mouseY: Int, mouseButton: Int): Boolean {
        return false
    }

    override fun getIngredients(ingredients : IIngredients?) {
    }
}
